### PR TITLE
[RAPTOR-4480] add support for loading arbitrary Java predictor

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Currently all our Java predictors are public and are packaged into a  jar deployed with DRUM.
This allows specify a class path and a class to load for external jar predictor,

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py [hunk 1]
```diff
@@ -191,7 +194,8 @@ def _run_java_server_entry_point(self):
                 java_cp,
                 JavaPredictor.JAVA_COMPONENT_ENTRY_POINT_CLASS,
                 "--class-name",
-                JavaPredictor.java_class_by_ext[self.model_artifact_extension],
+                self._custom_predictor_class
```
Comment (liororama) [inline custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py:197]: I suggest to add a log message (info) that a custom predictor class was given and being used.

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py [hunk 2]
```diff
@@ -176,8 +177,10 @@ def _stop_py4j(self):
         self._cleanup()
 
     def _run_java_server_entry_point(self):
-
         java_cp = ":".join(self._jar_files)
+        custom_class_path = os.environ.get(EnvVarNames.DRUM_JAVA_CUSTOM_CLASS_PATH)
+        if custom_class_path is not None:
+            java_cp = "{}:{}".format(java_cp, custom_class_path)
```
Comment (liororama) [inline custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py:183]: A log message here would be good to. To show the content of the custom class path.

**Review by liororama — APPROVED** [review]

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py (hunk 2 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py:183]: Done

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py (hunk 1 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py]: Done

**Review by meltzg — CHANGES_REQUESTED** [review]
needs a test

**Comment by yakov-g** [comment]
jarvis test functional_inference_model_templates

**Comment by yakov-g** [comment]
> needs a test

added

**Review by meltzg — APPROVED** [review]

**Comment by yakov-g** [comment]
jarvis please test 0-all-check-status

**Comment by yakov-g** [comment]
jarvis please test black

**Comment by yakov-g** [comment]
jarvis please test 0-all-check-status

**Pull request merged**